store/file/ods.go

```diff
@@ -0,0 +1,280 @@
+		return nil, fmt.Errorf("syncing file: %w", err)
+	}
+	// TODO: fill ods field with data from eds
+	return &OdsFile{
+		path: path,
+		fl:   f,
+		hdr:  h,
+	}, nil
+}
+
+func writeOdsFile(w io.Writer, h *headerV0, eds *rsmt2d.ExtendedDataSquare) error {
+	err := writeHeader(w, h)
+	if err != nil {
+		return err
+	}
+
+	for _, shr := range eds.FlattenedODS() {
+		if _, err := w.Write(shr); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Size returns square size of the Accessor.
+func (f *OdsFile) Size(context.Context) int {
+	return f.size()
+}
+
+func (f *OdsFile) size() int {
+	return int(f.hdr.squareSize)
+}
+
+// Close closes the file.
+func (f *OdsFile) Close() error {
+	return f.fl.Close()
+}
+
+// Sample returns share and corresponding proof for row and column indices. Implementation can
+// choose which axis to use for proof. Chosen axis for proof should be indicated in the returned
+// Sample.
+func (f *OdsFile) Sample(ctx context.Context, rowIdx, colIdx int) (shwap.Sample, error) {
+	// Sample proof axis is selected to optimize read performance.
+	// - For the first and second quadrants, we read the row axis because it is more efficient to read
+	//   single row than reading full ods to calculate single column
+	// - For the third quadrants, we read the column axis because it is more efficient to read single
+	//   column than reading full ods to calculate single row
+	// - For the fourth quadrant, it does not matter which axis we read because we need to read full ods
+	//   to calculate the sample
+	axisType, axisIdx, shrIdx := rsmt2d.Row, rowIdx, colIdx
+	if colIdx < f.size()/2 && rowIdx >= f.size()/2 {
+		axisType, axisIdx, shrIdx = rsmt2d.Col, colIdx, rowIdx
+	}
+
+	axis, err := f.axis(ctx, axisType, axisIdx)
+	if err != nil {
+		return shwap.Sample{}, fmt.Errorf("reading axis: %w", err)
+	}
+
+	return shwap.SampleFromShares(axis, axisType, axisIdx, shrIdx)
+}
+
+// AxisHalf returns half of shares axis of the given type and index. Side is determined by
+// implementation. Implementations should indicate the side in the returned AxisHalf.
+func (f *OdsFile) AxisHalf(ctx context.Context, axisType rsmt2d.Axis, axisIdx int) (eds.AxisHalf, error) {
+	// read axis from file if axisis row and from top half of the square or if axis is column and from
+	// left half of the square
+	if axisIdx < f.size()/2 {
+		shares, err := f.readAxisHalf(axisType, axisIdx)
+		if err != nil {
+			return eds.AxisHalf{}, fmt.Errorf("reading axis half: %w", err)
+		}
+		return eds.AxisHalf{
+			Shares:   shares,
+			IsParity: false,
+		}, nil
+	}
+
+	// if axis is from the second half of the square, read full ods and compute the axis half
+	err := f.readOds()
+	if err != nil {
+		return eds.AxisHalf{}, err
+	}
+
+	shares, err := f.ods.computeAxisHalf(ctx, axisType, axisIdx)
+	if err != nil {
+		return eds.AxisHalf{}, fmt.Errorf("computing axis half: %w", err)
+	}
+	return eds.AxisHalf{
+		Shares:   shares,
+		IsParity: false,
+	}, nil
```

Comment:
I am a bit concerned that this gonna recompute halfs twice:
* computeAxisHalf: square -> to parity half
* AxisHalf: parity half -> original
  * When original is already in the square

Reply:
This comment is also wrong and based on a misunderstanding of computeAxisHalf works